admin_guide/advanced_scheduling/node_affinity.adoc

```diff
@@ -0,0 +1,283 @@
+[[admin-guide-sched-affinity]]
+= Using Node Affinity
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+_Node affinity_ is a set of rules used by the scheduler to determine which nodes are selected for a pod. The rules are defined using custom xref:../architecture/core_concepts/pods_and_services.html#labels[labels on nodes] and selectors specified in pods.
+
+Node affinity allows you to constrain which nodes your pod is eligible to be scheduled on using labels on the node.
+
+There are two types of node affinity rules, _required_ and _preferred_.
+
+Required rules *must* be met before a pod can be scheduled on a node. Preferred rules specify that, if the rule is met, the scheduler will try to enforce the rules, but will not guarantee enforcement.
+
+[NOTE]
+====
+If labels on a node change at runtime that results in an node affinity rule on a pod no longer being met, the pod will still continue to run on the node.
+====
+
+[[admin-guide-sched-affinity-config]]
+== Configuring Node Infinity
+
+Node affinity is configured through the pod specification file. You can specify a xref:admin-guide-sched-affinity-config-req[required rule], a xref:admin-guide-sched-affinity-config-pref[preferred rule], or both. If you specify both, the node must first meet the required rule, then meet the preferred.
+
+The following example is a node specification with a required rule that the pod can only be placed on a node with a label whose key is `e2e-az-NorthSouth` and whose value is either `e2e-az-North` or `e2e-az-South`.
+
+.Example pod config file with a node affinity required rule
+====
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: with-node-affinity
+spec:
+  affinity:
+    nodeAffinity: <1>
+      requiredDuringSchedulingIgnoredDuringExecution: <2>
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: e2e-az-NorthSouth <3>
+            operator: In <4>
+            values:
+            - e2e-az-North <3>
+            - e2e-az-South <3>
+  containers:
+  - name: with-node-affinity
+    image: gcr.io/google_containers/pause:2.0
+----
+====
+<1> The stanza to configure node affility.
+<2> Defines a required rule.
+<3> The key/value pair (label) that must be matched to apply the rule.
+<4> The xref:../../rest_api/openshift_v1.html#unversioned-labelselectorrequirement[operator] represents the relationship between the key and a set of values.  Can be `In`, `NotIn`, `Exists`, or `DoesNotExist`, `lt` or `gt`.
```

Comment:
1. not the relationship between the key and a set of values.
it is the ralationship between ( (the value of the key) on the node) and (the set of values in matchexpressions).
2. it is case sensitive,  lt gt should be Lt Gt.